### PR TITLE
Task-25-Linkedin-Auth-Front-End-/-Back-End

### PR DESCRIPTION
**Pull request by melonholy**

### CODE REVIEW

File: src/containers/auth/login/index.js [hunk 1]
```diff
@@ -50,13 +63,12 @@ class Login extends Component {
...
 
-  handleGoogleResponse = response => {
-    this.props.googleLoginRequest(response);
-  };
-
   render() {
     const { email, password, errors } = this.state;
     const { modalStatus } = this.props;
+    const linkedInURL = `http://localhost:8080/api/users/auth/linkedin`;
+    const googleURL = `http://localhost:8080/api/users/auth/google`;
```
Comment (slavasgod) [inline src/containers/auth/login/index.js]: URL hardcode?

File: src/containers/auth/login/index.js [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
-class Login extends Component {
+class Login extends ReactQueryParams {
   state = {
     email: '',
     password: '',
     errors: {},
   };
 
+  componentWillMount() {
+    if (this.queryParams.tokenGoogle) {
```
Comment (slavasgod) [inline src/containers/auth/login/index.js]: `componentWillMount` is unsafe and deprecated, remove it

File: src/containers/auth/login/index.js [hunk 3]
```diff
@@ -1,22 +1,35 @@
-import React, { Component, Fragment } from 'react';
+import React, { Fragment } from 'react';
 import Modal from 'react-modal';
 import { connect } from 'react-redux';
 import { isEmpty } from 'lodash';
 import { FaTimes } from 'react-icons/fa';
-import GoogleLogin from 'react-google-login';
-import { loginRequest, googleLoginRequest } from '../../../actions/auth';
+import ReactQueryParams from 'react-query-params';
+import {
```
Comment (slavasgod) [inline src/containers/auth/login/index.js]: Fix imports order: base react libraries - custom react libraries - your components - styles. Separate every group by whitespace. Check it in all your files

File: src/containers/auth/styles.module.scss [hunk 4]
```diff
@@ -75,5 +75,46 @@
 }
 
 .googleButton {
-  margin: 5px !important;
+  margin: 5px 0;
+  width: 210px;
+  height: 40px;
+  border-width: 0;
+  background: white;
+  color: #737373;
```
Comment (slavasgod) [inline src/containers/auth/styles.module.scss]: magic colors?

File: src/containers/auth/styles.module.scss [hunk 5]
```diff
@@ -75,5 +75,46 @@
...
+  transition-timing-function: ease-in-out;
+  padding: 0;
+
+  &:focus,
+  &:hover {
+    box-shadow: 1px 4px 5px 1px rgba(0, 0, 0, 0.1);
+  }
+
+  &:active {
+    background-color: #e5e5e5;
```
Comment (slavasgod) [inline src/containers/auth/styles.module.scss]: magic color?

File: src/containers/auth/styles.module.scss [hunk 6]
```diff
@@ -75,5 +75,46 @@
...
+    box-shadow: none;
+    transition-duration: 10ms;
+  }
+
+  .googleButtonIcon {
```
Comment (slavasgod) [inline src/containers/auth/styles.module.scss]: we don't use camelCase in css

File: src/containers/auth/styles.module.scss [hunk 7]
```diff
@@ -75,5 +75,46 @@
...
+    box-sizing: border-box;
+  }
+
+  .googleButtonText {
+    display: inline-block;
+    vertical-align: middle;
+    padding: 0 24px;
+    font-size: 14px;
+    font-weight: bold;
+    font-family: 'Roboto', arial, sans-serif;
```
Comment (slavasgod) [inline src/containers/auth/styles.module.scss]: We have variables for fonts. Please use numbers for `font-weight`

File: src/containers/auth/login/index.js [hunk 8]
```diff
@@ -112,6 +165,7 @@ const mapStateToProps = state => ({});
 const mapDispatchToProps = {
   loginRequest,
   googleLoginRequest,
+  linkedInLoginRequest,
 };
 
 export default connect(
```
Comment (slavasgod) [inline src/containers/auth/login/index.js]: If you don't use mapStateToProps, replace it on `null`

File: package.json [hunk 9]
```diff
@@ -42,14 +42,17 @@
     "axios": "^0.19.0",
     "joi-browser": "^13.4.0",
     "jwt-decode": "^2.2.0",
+    "linkedin-login-for-react": "^1.2.3",
```
Comment (slavasgod) [inline package.json]: I didn't find place, where you used this library. If you didn't use this library - remove it. Also you don't use google login library

**Review by slavasgod — CHANGES_REQUESTED** [review]

File: src/containers/auth/login/index.js [hunk 10]
```diff
@@ -107,11 +163,12 @@ class Login extends Component {
   }
 }
 
-const mapStateToProps = state => ({});
+const mapStateToProps = null;
```
Comment (slavasgod) [inline src/containers/auth/login/index.js]: I mean pass null into connect function:
```
export default connect(
  null,
  mapDispatchToProps
)(Login);
```

File: src/containers/auth/login/index.js [hunk 11]
```diff
@@ -89,14 +104,55 @@ class Login extends Component {
...
+                      id="Shape"
+                      fill="#4285F4"
+                    />
+                    <path
+                      d="M81.4 222.2c7.8 22.9 22.8 43.2 42.6 57.1 12.4 8.7 26.6 14.9 41.4 17.9 14.6 3 29.7 2.6 44.4.1 14.6-2.6 28.7-7.9 41-16.2l59.1 45.9c-21.3 19.7-48 33.1-76.2 39.6-31.2 7.1-64.2 7.3-95.2-1-24.6-6.5-47.7-18.2-67.6-34.1-20.9-16.6-38.3-38-50.4-62 20.3-15.7 40.6-31.5 60.9-47.3z"
+                      fill="#34A853"
+                    />
+                  </svg>
+                </span>
+                <span className={styles.google_button_text}>
```
Comment (universezzz) [inline src/containers/auth/login/index.js:112]: Why did you use `<span>` for google and linkedin auth? This tag made to wrap small blocks or decorative elements of a web page. Please, use `<button>`.

File: src/containers/auth/login/index.js [hunk 12]
```diff
@@ -17,6 +25,19 @@ class Login extends Component {
     errors: {},
   };
 
+  componentDidMount() {
+    const { location, history } = this.props;
```
Comment (universezzz) [inline src/containers/auth/login/index.js:27]: if you destructed location and history, why did you not destruct googleLoginRequest and linkedInLoginRequest from props?

File: src/containers/auth/login/index.js [hunk 13]
```diff
@@ -92,14 +110,55 @@ class Login extends Component {
...
-            <GoogleLogin
-              clientId={process.env.CLIENT_ID}
-              onSuccess={this.handleGoogleResponse}
-              buttonText="Login"
-              className={styles.googleButton}
-              cookiePolicy={'single_host_origin'}
-            />
-            ,
+            <a href={googleURL}>
+              <div className={styles.google_button}>
```
Comment (OlegPanasyuk) [inline src/containers/auth/login/index.js:108]: Use destruction for styles

File: src/containers/auth/login/index.js [hunk 14]
```diff
@@ -51,15 +72,12 @@ class Login extends Component {
...
-  handleGoogleResponse = response => {
-    const { googleLoginRequest, onModalClose } = this.props;
-    googleLoginRequest(response);
-    onModalClose(false);
-  };
-
   render() {
     const { email, password, errors } = this.state;
     const { modalStatus } = this.props;
+    const linkedInURL = `api/users/auth/linkedin`;
```
Comment (universezzz) [inline src/containers/auth/login/index.js:72]: I don't know, but maybe, isn't a good idea, to make variables for google and linkedin urls? You can insert string presentetion of url to the href parameter of link right away. <`a href={'api/users/auth/google'}>`. First of all, you don't allocate memory for variable.

File: src/containers/auth/register/index.js [hunk 15]
```diff
@@ -21,6 +22,19 @@ class Register extends Component {
     errors: {},
   };
 
+  componentDidMount() {
```
Comment (universezzz) [inline src/containers/auth/register/index.js:28]: if you destructed location and history, why did you not destruct googleLoginRequest and linkedInLoginRequest from props?

**Review by universezzz — CHANGES_REQUESTED** [review]

File: src/containers/auth/login/index.js [hunk 16]
```diff
@@ -17,20 +18,23 @@ import styles from '../styles.module.scss';
...
+    const { location, history } = this.props;
+    const parsed = queryString.parse(location.search);
+    if (parsed.tokenGoogle) {
+      this.props.googleLoginRequest(parsed.tokenGoogle);
+      history.push('/');
     }
-    if (this.queryParams.tokenLinkedin) {
-      this.props.linkedInLoginRequest(this.queryParams.tokenLinkedin);
+    if (parsed.tokenLinkedin) {
+      this.props.linkedInLoginRequest(parsed.tokenLinkedin);
```
Comment (melonholy) [inline src/containers/auth/login/index.js]: make it as one united token

File: src/containers/auth/register/index.js [hunk 17]
```diff
@@ -21,6 +22,19 @@ class Register extends Component {
...
+    const { location, history } = this.props;
+    const parsed = queryString.parse(location.search);
+    if (parsed.tokenGoogle) {
```
Comment (melonholy) [inline src/containers/auth/register/index.js]: make HOC for google/linkedin login and provide as children needed inputs and button. CC @universezzz

File: src/containers/auth/register/index.js [hunk 18]
```diff
@@ -21,6 +22,19 @@ class Register extends Component {
...
+      this.props.googleLoginRequest(parsed.tokenGoogle);
+      history.push('/');
+    }
+    if (parsed.tokenLinkedin) {
```
Comment (melonholy) [inline src/containers/auth/register/index.js]: also get rid of if's

File: src/containers/auth/register/index.js [hunk 19]
```diff
@@ -144,12 +160,55 @@ class Register extends Component {
...
-            <GoogleLogin
-              clientId={process.env.CLIENT_ID}
-              buttonText="Login"
-              className={styles.googleButton}
-              cookiePolicy={'single_host_origin'}
-            />
+            <a href={googleURL}>
+              <div className={styles.google_button}>
+                <span className={styles.google_button_icon}>
+                  <svg viewBox="0 0 366 372" xmlns="http://www.w3.org/2000/svg">
```
Comment (melonholy) [inline src/containers/auth/register/index.js]: add svg loader to webpack and create svg file -> import him here as component

File: src/containers/auth/login/index.js [hunk 20]
```diff
@@ -92,14 +104,26 @@ class Login extends Component {
...
+                </span>
+                <span className={styles.google_button_text}>
+                  Sign in with Google
+                </span>
+              </div>
+            </a>
+            <a href={linkedInURL}>
+              <div className={styles.google_button}>
+                <span className={styles.google_button_icon}>
+                  <LinkedIn />
```
Comment (melonholy) [inline src/containers/auth/login/index.js]: add icon to name

File: src/services/auth.js [hunk 21]
```diff
@@ -1,14 +1,21 @@
...
 import jwt_decode from 'jwt-decode';
+import axios from 'axios';
 import setAuthToken from './setAuthToken';
 
 const config = {
   headers: { 'Content-Type': 'application/json' },
 };
 
+export const socialLoginRequest = res => {
+  const decoded = jwt_decode(res.payload);
```
Comment (melonholy) [inline src/services/auth.js:10]: remove jwt-decode lib from front-end. send get request for user data with this token.

File: src/services/auth.js [hunk 22]
```diff
@@ -1,14 +1,21 @@
...
+  localStorage.setItem('token', res.payload);
```
Comment (melonholy) [inline src/services/auth.js:11]: write some wrapper to local storage

File: src/services/auth.js [hunk 23]
```diff
@@ -1,14 +1,21 @@
...
+  setAuthToken(res.payload);
+  return decoded;
```
Comment (melonholy) [inline src/services/auth.js:14]: newline before return

File: src/services/auth.js [hunk 24]
```diff
@@ -1,14 +1,21 @@
...
 import setAuthToken from './setAuthToken';
 
 const config = {
   headers: { 'Content-Type': 'application/json' },
 };
 
+export const socialLoginRequest = res => {
+  const decoded = jwt_decode(res.payload);
+  localStorage.setItem('token', res.payload);
+  setAuthToken(res.payload);
```
Comment (melonholy) [inline src/services/auth.js:12]: rewrite token flow

File: package.json [hunk 25]
```diff
@@ -34,20 +34,24 @@
...
+    "svg-react-loader": "^0.4.6",
     "webpack": "^4.39.1",
     "webpack-cli": "^3.3.6",
     "webpack-dev-server": "^3.7.2"
   },
   "dependencies": {
+    "@svgr/webpack": "^4.3.2",
     "axios": "^0.19.0",
+    "babel-plugin-named-asset-import": "^0.3.3",
+    "history": "^4.9.0",
```
Comment (slavasgod) [inline package.json:46]: `@svgr/webpack` and `babel-plugin-named-asset-import` should be in devDependencies

**Review by slavasgod — CHANGES_REQUESTED** [review]

**Review by OlegPanasyuk — APPROVED** [review]

**Review by slavasgod — APPROVED** [review]

**Pull request merged**